File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int n=nums.size();
        //            sum,ind
        unordered_map<int,int>  mpp;

        for(int i=0;i<n;i++){
           
            int rem=target-nums[i];
            if(mpp.count(rem)){
                return {i,mpp[rem]};
            }
            mpp[nums[i]]=i;
        }

        return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> indMap=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int rem=target-nums[i];
            if(indMap.containsKey(rem)){
                return new int[]{indMap.get(rem), i};
            }
            indMap.put(nums[i],i);
        }
        return null;
    }
}